client-dotnet/src/Client.Util/SemaphoreLocker.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Client.Util {
+    public class SemaphoreLocker {
```

Comment:
add some documentation explaining what this class is for

Reply:
Some comments added